psycop/common/model_training_v2/trainer/data/data_filters/geography.py

```diff
@@ -0,0 +1,47 @@
+from collections.abc import Collection
+from typing import Literal
+
+import polars as pl
+
+from psycop.common.data_inspection.visits_by_hospital_units.make_geographical_split import (
+    GEOGRAPHICAL_SPLIT_PATH,
+)
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
+from psycop.common.model_training_v2.trainer.base_dataloader import BaselineDataLoader
+from psycop.common.model_training_v2.trainer.data.data_filters.base_data_filter import (
+    BaselineDataFilter,
+)
+
+
+@BaselineRegistry.data_filters.register("geography_data_filter")
+class GeographyDataFilter(BaselineDataFilter):
+    def __init__(
+        self,
+        regions: Collection[Literal["vest", "midt", "øst"]],
+        id_col_name: str,
+        timestamp_col_name: str,
+    ):
+        self.splits = regions
+        self.id_col_name = id_col_name
+        self.timestamp_col_name = timestamp_col_name
+
+    def apply(self, dataloader: BaselineDataLoader) -> pl.LazyFrame:
+        """Filter the dataloader to only include ids from the desired regions
+        and remove prediction times after a move to a different region"""
+        filtered_geography_id_df = self._load_and_prepare_geography_df()
```

Comment:
Could we make it more generic, so this loading can be moved to the function signature? (Would make it easier to understand IME).